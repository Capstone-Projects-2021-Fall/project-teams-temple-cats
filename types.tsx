/**
 * Learn more about using TypeScript with React Navigation:
 * https://reactnavigation.org/docs/typescript/
 */

import { BottomTabScreenProps } from '@react-navigation/bottom-tabs';
import { CompositeScreenProps, NavigatorScreenParams } from '@react-navigation/native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { LatLng } from 'react-native-maps';

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}

export type RootStackParamList = {
  Root: NavigatorScreenParams<RootTabParamList> | undefined;
  Announcements: undefined;
  CatForm: undefined;
  Facebook: undefined;
  Settings: undefined;
  Rewards: undefined;
  NotFound: undefined;
  Cat: {cat: Cat};
  FeedingStations: {title: String, info: any};
  ReportedPosts: undefined;
  DownvotedPosts: undefined;
};

export type RootStackScreenProps<Screen extends keyof RootStackParamList> = NativeStackScreenProps<
  RootStackParamList,
  Screen
>;

export type RootTabParamList = {
  Home: undefined;
  Resources: undefined;
  Leaderboard: undefined;
  Account: undefined;
  Facebook: undefined;
};

export type RootTabScreenProps<Screen extends keyof RootTabParamList> = CompositeScreenProps<
  BottomTabScreenProps<RootTabParamList, Screen>,
  NativeStackScreenProps<RootStackParamList>
>;

export type Report = {
  reportID: string;
  catID: string;
  accountID: string;
  reason: string;
};

export type Comment = {
  commentID: string;
  content: string;
  accountID: string;
  reports: Report[];
};

export type Announcement = {
  AnnouncementID: string;
  Time: Date;
  Content: string;
  Type: string;
  AccountID: string;
};

export type Account = {
  accountID: string;
  modStatus: boolean;
  banned: boolean;
  points: number;
};

export type FeedingStation = {
  StationID: string;
  Media: string;
  Comments: string;
};

export type Cat = {
  catID: string;
  media: string;
  name: string;
  color: string;
  eyeColor: string;
  kitten: boolean;
  healthy: boolean;
  friendly: boolean;
  comments: string;
  location: LatLng;
  date: Date;
  time: Date;
  votes: number;
  accountID: string;
  commentList: Comment[];
  reports: Report[];
};

export type FeedingStations = {
 title: String;
 info: String;
 latitude: any,
 longitude: any,
 street: String,
};



export type User = {
  displayName: string | null | undefined;
  accountID: string | undefined;
  email: string | null | undefined;
  photo: string | null | undefined;
  points: 0;
  posts: 0;
  modStatus: 1 | 2 | 3; // user, feeder, moderator
  banStatus: true | false;
};
