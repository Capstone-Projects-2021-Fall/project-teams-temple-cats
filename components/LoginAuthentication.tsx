import * as React from "react";
import * as WebBrowser from "expo-web-browser";
import * as Facebook from "expo-auth-session/providers/facebook";
import { ResponseType } from "expo-auth-session";
import firebase from "../utils/firebase";
import { Button } from "react-native";

WebBrowser.maybeCompleteAuthSession();

//Request response token from user

/**
 * Function that renders the Login Authentication component with a button to log in via Facebook.
 * @component
 * @returns {JSX.Element} JSX element of the login
 */
export default function LoginAuthentication() {
  /**
   * Facebook authorization request's loaded request object.
   * @constant {FacebookAuthRequest} request
   * @memberof LoginAuthentication
   */
  /**
   * Facebook authorization request's response object.
   * @constant {AuthSessionResult} response
   * @memberof LoginAuthentication
   */
  /**
   * Facebook authorization request's prompt method.
   * @method promptAsync
   * @memberof LoginAuthentication
   */
  const [request, response, promptAsync] = Facebook.useAuthRequest({
    responseType: ResponseType.Token,
    clientId: "562935831789483",
  });

  React.useEffect(() => {
    if (response?.type === "success") {
      const { access_token } = response.params;
      const credential =
        firebase.auth.FacebookAuthProvider.credential(access_token);
      firebase.auth().signInWithCredential(credential);
      console.log("User added");
    }
  }, [response]);

  return (
    <Button
      disabled={!request}
      color="#8b0000"
      title="Login"
      onPress={() => {
        promptAsync();
      }}
    />
  );
}
