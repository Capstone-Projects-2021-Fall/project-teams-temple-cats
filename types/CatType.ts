export default interface CatType {
    key?: string | null,
    title: string,
    description: string,
    published?: boolean,
  }