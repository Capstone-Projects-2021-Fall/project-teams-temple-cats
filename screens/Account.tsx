import * as React from "react";
import { StyleSheet, Button } from "react-native";

import EditScreenInfo from "../components/EditScreenInfo";
import { Text, View } from "../components/Themed";
import { RootTabScreenProps } from "../types";
import firebase from "firebase";
import { AuthContext } from "../context/FirebaseAuthContext";

export default function AccountScreen({
  navigation,
}: RootTabScreenProps<"Account">) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Account</Text>
      <View
        style={styles.separator}
        lightColor="#eee"
        darkColor="rgba(255,255,255,0.1)"
      />
      <Button
        color="#8b0000"
        title="Sign Out"
        onPress={() => {
          signOut();
        }}
      />
      <EditScreenInfo path="/screens/Accounts.tsx" />
    </View>
  );
}

const signOut = async () => {
  firebase.auth().signOut();
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
});
