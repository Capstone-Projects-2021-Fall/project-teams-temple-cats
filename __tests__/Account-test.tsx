import * as React from 'react';
import { render, fireEvent} from '@testing-library/react-native';
import Account from '../screens/Account';
import LoginAuthentication from '../components/LoginAuthentication';

import { RootTabScreenProps } from '../types';


it("renders default elements", () => {
    render(<Account/>);
  });

  it("renders correct display element", () => {
     const {getAllByText} = render(<Account/>);
     expect(getAllByText('Account').length).toBe(1);
  });