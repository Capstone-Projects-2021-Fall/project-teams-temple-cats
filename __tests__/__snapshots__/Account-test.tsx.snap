// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Account
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        },
      ]
    }
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#8b0000",
            },
          ]
        }
      >
        Sign Out
      </Text>
    </View>
  </View>
</View>
`;
